yktrace/src/tir.rs

```diff
@@ -617,11 +627,34 @@ impl TirOp {
 }
 
 #[cfg(test)]
-mod tests {
+pub mod test_helpers {}
+
+#[cfg(test)]
+pub mod tests {
     use super::TirTrace;
-    use crate::{sir::SIR, start_tracing, TracingKind};
+    use crate::{sir::SIR, start_tracing, trace_debug, TracingKind};
+    use fm::FMBuilder;
+    use regex::Regex;
     use test::black_box;
 
+    /// Fuzzy matches the textual TIR for the trace `tt` with the pattern `ptn`.
+    pub fn assert_tir(ptn: &str, tt: &TirTrace) {
+        let ptn_re = Regex::new(r"%.+?\b").unwrap(); // Names are words prefixed with `%`.
+        let text_re = Regex::new(r"\$?.+?\b").unwrap(); // Any word optionally prefixed with `$`.
+        let matcher = FMBuilder::new(ptn)
+            .unwrap()
+            .name_matcher(Some((ptn_re, text_re)))
+            .distinct_name_matching(true)
+            .build()
+            .unwrap();
+
+        let res = matcher.matches(&format!("{}", tt));
+        if let Err(e) = res {
+            eprintln!("{}", e); // Visible when tests run with --nocapture.
```

Comment:
Doesn't `panic` print this either way, `capture` or `nocapture`? Why do we need the `eprint`?

Reply:
I ran some experiments.

the panic argument is always printed, but it doesn't print the error in this case. It says:
```
thread 'tir::tests::trace_debug_tir' panicked at 'Box<Any>', yktrace/src/tir.rs:654:13
```

i.e. not very useful. Will fix.

Reply:
`panic!(e)` will directly throw `Box::new(e) as Box<Any>` instead of `Box::new(format!("{}", e)) as Box<Any>` like `panic!("{}", e)` does.

Reply:
Thanks. Fixed in c38fa973f372c47022b3c66d122b5bc334b2d2af